### ISSUE TITLE
Possible inconsistency with default `config.source` value

### ISSUE THREAD

**Issue by leggomuhgreggo** [DataDog/dd-sdk-reactnative#211]
Hello!

I reached out to support a while back and wanted to share the findings.

Here's the background

>>We're working on a cross-platform React Native app -- so web and native report to the same RUM app -- and I was trying to figure out the best way to distinguish between >>the native app and web.
>>
>>Currently there's a "mobile" facet which is great! BUT it includes mobile web -- and I am trying to get *just* the native app sessions.
>>
>>There's also "OS" facet, but that includes mobile web results, as well.
>>
>>So far the best I've been able to figure out is to add `-@browser.name:*` to the search, to exclude any sessions with a defined browser.
>>
>>But perhaps I have overlooked something important?
>>
>>My hope is that we can figure out a way to quickly toggle between native and web platforms -- perhaps a "Platform" facet?

>After taking a look at your case, I think that using the source facet can achieve your requirement, would you be able to confirm that?

The `source` property definitely sounds like what Im looking for -- but it appears as tho android is being grouped as "react-native" [see: screenshot]

![image](https://user-images.githubusercontent.com/2213636/171523210-4cd5fef9-28d6-473f-b97e-cddf9df5dc35.png)

I looked around the codebase and it looks like the default is being defined as `react-native` [here](https://github.com/DataDog/dd-sdk-reactnative/blob/b3d9f1666742bdedb144f3f17d0669367be34fb8/packages/core/src/DdSdkReactNative.tsx#L63-L64)

So I guess the actual question is: "why is `ios` appearing as the source at all?"*

I think the solution is probably just to manually define the source property -- but I thought I'd check here to see if the expected behavior should be to autodetect the platform by default.

Thanks!

---

#### Version

Been staying up to date withe release candidates:

```
    "@datadog/browser-rum": "3.2.0",
    "@datadog/mobile-react-native": "1.0.0-rc8",
    "@datadog/mobile-react-navigation": "1.0.0-rc8",
```

**Comment by 0xnm** [comment]
Hello @leggomuhgreggo. Thanks for opening the issue. It was actually a bug in a native part of our SDK which was fixed recently in https://github.com/DataDog/dd-sdk-ios/pull/832/commits/ae560c22ab8ed75d3c14e483135f20b922a4baca. The fix should be available in our latest React Native SDK version 1.0.0-rc8 release.

Can you please check that you don't have this issue for your app version using `1.0.0-rc8` and it happens only for app versions coming with earlier React Native SDK versions?

**Comment by leggomuhgreggo** [comment]
Looks good! Thanks for the context.

![image](https://user-images.githubusercontent.com/2213636/171602923-8aa8ba24-6607-4707-b111-e3737a562e01.png)
